src/config/machines/example-grant-machine.js

```diff
@@ -171,9 +174,29 @@ export const exampleGrantMachine = createMachine({
         ...pageUIConfig.consent
       }
     },
-
     confirmation: {
-      type: 'final',
+      initial: 'submitting',
+      states: {
+        submitting: {
+          invoke: {
+            // eslint-disable-next-line jsdoc/require-jsdoc
+            src: (context) => {
+              if (!context.hasSubmitted) {
+                handleSubmission(context);
+                context.hasSubmitted = true;
```

Comment:
The handleSubmission function updates context.hasSubmitted directly. However, state machine context should be immutable to maintain consistency. Use an action to update hasSubmitted

Reply:
Sth like the below:

```
actions: {
  markSubmissionComplete: assign({
    hasSubmitted: () => true
  })
}
```

```
invoke: {
  src: async (context) => {
    await handleSubmission(context);
  },
  onDone: {
    target: 'success',
    actions: ['markSubmissionComplete']
  },
  onError: {
    actions: (_, event) => console.error('Submission failed:', event.data)
  }
}
```